pkg/conversion/mapi2capi/powervs_test.go

```diff
@@ -0,0 +1,142 @@
+			InfrastructureName: "sample-cluster-name",
+		},
+	}
+
+	// We need to add error handling for the providerSpec issues before we can enable the following tests.
+	PIt("should be able to convert a MAPI Power VS Machine to a CAPI Machine", func() {
+		// Convert a MAPI Machine to a CAPI Core Machine + a CAPI InfraMachine.
+		capiMachine, capiInfraMachine, warns, err :=
+			FromPowerVSMachineAndInfra(powerVSMAPIMachine, &infra).ToMachineAndInfrastructureMachine()
+		fmt.Print(capiMachine)
+		Expect(err).ToNot(HaveOccurred(), "should have been able to convert providerSpec to MachineTemplateSpec")
+		Expect(warns).To(BeEmpty(), "should have not warned while converting providerSpec to MachineTemplateSpec")
+		Expect(capiMachine).To(Not(BeNil()), "should not have a nil CAPI Machine")
+		Expect(capiInfraMachine).To(Not(BeNil()), "should not have a nil CAPI InfrastructureMachine")
+	})
+
+	It("should be able to convert a MAPI Power VS MachineSet to a CAPI MachineSet", func() {
+		// Convert a MAPI MachineSet to a CAPI Core MachineSet + a CAPI InfraMachineTemplateSpec.
+		capiMachineSet, capiInfraMachineTemplate, warns, err :=
+			FromPowerVSMachineSetAndInfra(powerVSMAPIMachineSet, &infra).ToMachineSetAndMachineTemplate()
+		Expect(err).ToNot(HaveOccurred(), "should have been able to convert MAPI MachineSet to CAPI MachineSet")
+		Expect(warns).To(BeEmpty(), "should have not warned while converting MAPI MachineSet to CAPI MachineSet")
+		Expect(capiMachineSet).To(Not(BeNil()), "should not have a nil CAPI MachineSet")
+		Expect(capiInfraMachineTemplate).To(Not(BeNil()), "should not have a nil CAPI MachineTemplate")
+	})
+
+})
+
+// getPowerVSProviderSpec builds and returns PowerVSProviderConfig
+func getPowerVSProviderSpec() *mapiv1.PowerVSMachineProviderConfig {
+	// TODO: May be we can add this in machine builder?
+	return &mapiv1.PowerVSMachineProviderConfig{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{},
+		UserDataSecret: &mapiv1.PowerVSSecretReference{
+			Name: "worker-user-data",
+		},
+		CredentialsSecret: &mapiv1.PowerVSSecretReference{
+			Name: "powervs-credentials",
+		},
+		ServiceInstance: mapiv1.PowerVSResource{
+			Type:  mapiv1.PowerVSResourceTypeID,
+			ID:    ptr.To("1234"),
+			Name:  nil,
+			RegEx: nil,
+		},
+		Image: mapiv1.PowerVSResource{
+			Type: mapiv1.PowerVSResourceTypeName,
+			Name: ptr.To("rhcos-ipi-sa04-418nig-jqw97"),
+		},
+		Network: mapiv1.PowerVSResource{
+			Type:  mapiv1.PowerVSResourceTypeRegEx,
+			RegEx: ptr.To("^DHCPSERVER.*ipi-sa04-418nig-jqw97.*_Private$"),
+		},
+		KeyPairName:   "ipi-sa04-418nig-jqw97-key",
+		SystemType:    "s922",
+		ProcessorType: mapiv1.PowerVSProcessorTypeShared,
+		Processors:    intstr.FromString("2"),
+		MemoryGiB:     32,
+	}
+}
+
+// rawExtensionFromPowerVSProviderSpec marshals the machine provider spec.
+func rawExtensionFromPowerVSProviderSpec(spec *mapiv1.PowerVSMachineProviderConfig) (*runtime.RawExtension, error) {
```

Comment:
This keeps popping up, we need a generic util for this

Reply:
Just changed the param type to interface{}